src/main/java/org/jenkinsci/plugins/bigpanda/BigpandaGlobalNotifier.java

```diff
@@ -0,0 +1,127 @@
+package org.jenkinsci.plugins.bigpanda;
+
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+
+import hudson.Extension;
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.listeners.RunListener;
+import hudson.util.FormValidation;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+
+@Extension
+public class BigpandaGlobalNotifier extends RunListener<Run<?, ?>> implements Describable<BigpandaGlobalNotifier> {
+
+    private Run<?,?> build;
+    private JSONObject change;
+
+
+    @Override
+    public void onCompleted(Run<?, ?> run, TaskListener listener) {
+        publish(run, listener);
+    }
+
+    @Override
+    public void onStarted(Run<?, ?> run, TaskListener listener) {
+        publish(run, listener);
+    }
+
+    public Descriptor<BigpandaGlobalNotifier> getDescriptor() {
+        return getDescriptorImpl();
+    }
+
+    public DescriptorImpl getDescriptorImpl() {
+        return (DescriptorImpl) Jenkins.getInstance().getDescriptorOrDie(BigpandaGlobalNotifier.class);
+    }
+
+    private void publish(Run<?, ?> r, TaskListener listener) {
+
+        this.build = r;
+        ChangesBuilder changeBuilder = new ChangesBuilder(this.build);
+
+        Descriptor<BigpandaGlobalNotifier> descriptor = getDescriptor();
+
+        change = changeBuilder.create();
+
+        BigpandaApiWrapper bpApi = new BigpandaApiWrapper( ((DescriptorImpl) descriptor).getBigpandaApiKey(), ((DescriptorImpl) descriptor).getBigpandaAppKey());
+
+        try {
+            bpApi.notifyBigPanda(change);
+
+            listener.getLogger().println("BigPanda Notification Success");
+        } catch (Exception e) {
+            listener.getLogger().println("BigPanda Notification Failed: " + e.getMessage());
+        }
+    }
+
+    @Extension
+    public static final class DescriptorImpl extends Descriptor<BigpandaGlobalNotifier> {
+        private String bigpandaApiKey;
+        private String bigpandaAppKey;
+
+        public DescriptorImpl() {
+            load();
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Global BigPanda Notifications";
+        }
+
+        @Override
+        public boolean configure(StaplerRequest req, JSONObject formData) {
+            req.bindJSON(this, formData);
+
+            save();
+            return true;
+        }
+
+        /**
+         * 
+         * @return Bigpanda Api Key
+         */
+        public String getBigpandaApiKey() {
+            return this.bigpandaApiKey;
+        }
+
+        /**
```

Comment:
Can we add an optional field for the url?

Reply:
We absolutely can and I did debate this. 

What is your thought process with adding the URL?

My concern is that we are trying to make this a general solution that goes to the webhook we designed for this integration. So if we introduce the customization to change the endpoint, it no longer becomes a general solution